csharp/source/autowrap/csharp/csharp.d

```diff
@@ -0,0 +1,1061 @@
+    rangeDef.functions ~= "        [DllImport(\"%s\", EntryPoint = \"autowrap_csharp_slice_Wstring_Slice\", CallingConvention = CallingConvention.Cdecl)]".format(libraryName) ~ newline;
+    rangeDef.functions ~= "        internal static extern slice dlang_slice_Wstring_Slice(slice dslice, IntPtr begin, IntPtr end);" ~ newline;
+    rangeDef.functions ~= "        [DllImport(\"%s\", EntryPoint = \"autowrap_csharp_slice_Wstring_AppendValue\", CallingConvention = CallingConvention.Cdecl)]".format(libraryName) ~ newline;
+    rangeDef.functions ~= "        internal static extern slice dlang_slice_Wstring_AppendValue(slice dslice, slice value);" ~ newline;
+    rangeDef.functions ~= "        [DllImport(\"%s\", EntryPoint = \"autowrap_csharp_slice_Wstring_AppendSlice\", CallingConvention = CallingConvention.Cdecl)]".format(libraryName) ~ newline;
+    rangeDef.functions ~= "        internal static extern slice dlang_slice_Wstring_AppendSlice(slice dslice, slice array);" ~ newline;
+
+    //dstring
+    rangeDef.functions ~= "        [DllImport(\"%s\", EntryPoint = \"autowrap_csharp_slice_Dstring_Create\", CallingConvention = CallingConvention.Cdecl)]".format(libraryName) ~ newline;
+    rangeDef.functions ~= "        internal static extern slice dlang_slice_Dstring_Create(IntPtr capacity);" ~ newline;
+    rangeDef.functions ~= "        [DllImport(\"%s\", EntryPoint = \"autowrap_csharp_slice_Dstring_Get\", CallingConvention = CallingConvention.Cdecl)]".format(libraryName) ~ newline;
+    rangeDef.functions ~= "        internal static extern slice dlang_slice_Dstring_Get(slice dslice, IntPtr index);" ~ newline;
+    rangeDef.functions ~= "        [DllImport(\"%s\", EntryPoint = \"autowrap_csharp_slice_Dstring_Set\", CallingConvention = CallingConvention.Cdecl)]".format(libraryName) ~ newline;
+    rangeDef.functions ~= "        internal static extern void dlang_slice_Dstring_Set(slice dslice, IntPtr index, slice value);" ~ newline;
+    rangeDef.functions ~= "        [DllImport(\"%s\", EntryPoint = \"autowrap_csharp_slice_Dstring_Slice\", CallingConvention = CallingConvention.Cdecl)]".format(libraryName) ~ newline;
+    rangeDef.functions ~= "        internal static extern slice dlang_slice_Dstring_Slice(slice dslice, IntPtr begin, IntPtr end);" ~ newline;
+    rangeDef.functions ~= "        [DllImport(\"%s\", EntryPoint = \"autowrap_csharp_slice_Dstring_AppendValue\", CallingConvention = CallingConvention.Cdecl)]".format(libraryName) ~ newline;
+    rangeDef.functions ~= "        internal static extern slice dlang_slice_Dstring_AppendValue(slice dslice, slice value);" ~ newline;
+    rangeDef.functions ~= "        [DllImport(\"%s\", EntryPoint = \"autowrap_csharp_slice_Dstring_AppendSlice\", CallingConvention = CallingConvention.Cdecl)]".format(libraryName) ~ newline;
+    rangeDef.functions ~= "        internal static extern slice dlang_slice_Dstring_AppendSlice(slice dslice, slice array);" ~ newline;
```

Comment:
There's a huge amount of duplication or prefixes and suffixes in this (and elsewhere), surely there's a good way to make this very much smaller and less boilerplate-y?

How many things start with
```
`[DllImport("%s", EntryPoint = "autowrap_csharp_slice_`
```
or
```
`[DllImport("%s", EntryPoint = "autowrap_csharp_slice_%sstring_`.format(stringTypePrefix)
```
?

Reply:
This was a pretty big change but it should be up now.